### PR TITLE
Fix hyperscan build_chimera option

### PR DESCRIPTION
**Pull request by syoliver**
Specify library name and version:  **hyperscan/5.4.0**

Fix build_chimera option
Add components for cmake_find_package and cmake_find_package_multi generators

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/hyperscan/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -2,8 +2,17 @@ cmake_minimum_required(VERSION 3.1)
 project(PackageTest CXX)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
+
+find_package(hyperscan COMPONENTS hs REQUIRED)
+add_executable(hs_example hs_example.cpp)
+target_link_libraries(hs_example PRIVATE hyperscan::hs)
+set_property(TARGET hs_example PROPERTY CXX_STANDARD 11)
```
Comment (jwillikers) [inline recipes/hyperscan/all/test_package/CMakeLists.txt:10]: The C++ standard should be managed via Conan.

File: recipes/hyperscan/all/test_package/ch_example.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    ctx.ref = "123abcdefghijkl";
+    ch_compile_error_t *err = nullptr;
+    ch_database_t *db = nullptr;
+    ch_scratch_t *scratch = nullptr;
+
+    if (ch_compile("abc(\\w+)ghi", 0, CH_MODE_GROUPS, nullptr, &db, &err) != CH_SUCCESS) {
+        std::printf("Failed to compile database\n");
+        std::printf("%s\n", err->message);
+        retval = 1;
+        goto ret;
```
Comment (jwillikers) [inline recipes/hyperscan/all/test_package/ch_example.cpp:43]: Uh, `goto`?

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/hyperscan/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (syoliver) [reply recipes/hyperscan/all/test_package/CMakeLists.txt]: Removed

File: recipes/hyperscan/all/test_package/ch_example.cpp (hunk 2 above)
Comment (jwillikers) [reply recipes/hyperscan/all/test_package/ch_example.cpp]: Sorry, didn't mean to make more work for you. I understand just using the sample code and figure that's fine so long as the other maintainers don't have a problem with it.

**Review by jwillikers — APPROVED** [review]

File: recipes/hyperscan/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (syoliver) [reply recipes/hyperscan/all/test_package/CMakeLists.txt]: I don't find example of how to specify manually the cppstd in test_package. Maybe I don't search the right thing?

**Review by uilianries — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/hyperscan/all/conanfile.py [hunk 3]
```diff
@@ -95,5 +105,35 @@ def package(self):
...
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "libhs"
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package"] = "hyperscan"
+        self.cpp_info.names["cmake_find_package_multi"] = "hyperscan"
+
+        self.cpp_info.components["hs"].libs = ["hs"]
+        self.cpp_info.components["hs"].requires = ["boost::headers"]
+        self.cpp_info.components["hs"].names["cmake_find_package"] = "hs"
+        self.cpp_info.components["hs"].names["cmake_find_package_multi"] = "hs"
```
Comment (prince-chrismc) [inline recipes/hyperscan/all/conanfile.py:114]:
```suggestion
        self.cpp_info.components["hs"].names["cmake_find_package_multi"] = "hs"
        self.cpp_info.components["hs"].set_property("cmake_target_name", "hyperscan::hs")
```

File: recipes/hyperscan/all/conanfile.py [hunk 4]
```diff
@@ -95,5 +105,35 @@ def package(self):
...
+        self.cpp_info.components["hs"].set_property("pkg_config_name", "libhs")
+
+
+        self.cpp_info.components["hs_runtime"].libs = ["hs_runtime"]
+        self.cpp_info.components["hs_runtime"].names["cmake_find_package"] = "hs_runtime"
```
Comment (prince-chrismc) [inline recipes/hyperscan/all/conanfile.py:119]: no pkg config for this one?

File: recipes/hyperscan/all/conanfile.py [hunk 5]
```diff
@@ -95,5 +105,35 @@ def package(self):
...
+        self.cpp_info.components["hs_runtime"].names["cmake_find_package_multi"] = "hs_runtime"
```
Comment (prince-chrismc) [inline recipes/hyperscan/all/conanfile.py:120]:
```suggestion
        self.cpp_info.components["hs_runtime"].names["cmake_find_package_multi"] = "hs_runtime"
        self.cpp_info.components["hs"].set_property("cmake_target_name", "hyperscan::hs_runtime")
```

File: recipes/hyperscan/all/conanfile.py [hunk 6]
```diff
@@ -95,5 +105,35 @@ def package(self):
...
+
+
+        if self.options.build_chimera:
+            self.cpp_info.components["chimera"].libs = ["chimera"]
+            self.cpp_info.components["chimera"].requires = ["pcre::libpcre", "hs"]
+            self.cpp_info.components["chimera"].names["cmake_find_package"] = "chimera"
+            self.cpp_info.components["chimera"].names["cmake_find_package_multi"] = "chimera"
```
Comment (prince-chrismc) [inline recipes/hyperscan/all/conanfile.py:127]:
```suggestion
            self.cpp_info.components["chimera"].names["cmake_find_package_multi"] = "chimera"
        self.cpp_info.components["hs"].set_property("cmake_target_name", "hyperscan::chimera")
```

File: recipes/hyperscan/all/conanfile.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
 
 import os
```
Comment (prince-chrismc) [inline recipes/hyperscan/all/conanfile.py:4]:
```suggestion
import os

required_conan_version = ">=1.43.0"
```

**Review by prince-chrismc — COMMENTED** [review]
Minor comments about new properties, otherwise it looks good

File: recipes/hyperscan/all/conanfile.py (hunk 4 above)
Comment (syoliver) [reply recipes/hyperscan/all/conanfile.py:122]: Upstream doesn't provide a pkg config for this one, but I can add it in conanfile.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**